packages/runtime-core/__tests__/apiOptions.spec.ts

```diff
@@ -1704,5 +1706,104 @@ describe('api: options', () => {
         `Computed property "foo" is already defined in Data.`,
       ).toHaveBeenWarned()
     })
+
+    test('defineComponent with generic', async () => {
```

Comment:
I think the tests are related to types should be placed in `packages/dts-test/defineComponent.test-d.tsx`

Reply:
@edison1105 
thank you!
I moved it to defineComponent.test-d.tsx!

Reply:
It's still not quite right, you need to refer to the exist test cases in the `defineComponent.test-d.tsx` file to add new test cases. **Check the type only, do not test the render result**

Reply:
Modified! It's my insufficient understanding. Would you review the changes once more?

Reply:
those test cases seem not to be related generics. you should refer to this test case 
 https://github.com/vuejs/core/blob/d7d0371e74707ee601020f67de88e091cdae2673/packages-private/dts-test/defineComponent.test-d.tsx#L1273

Reply:
@edison1105 
My understanding of the tests was completely lacking. What do you think of this revision? 
I'd appreciate your review.